FWCore/Skeletons/test/runtests.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash -ex
+
+SCRAM_TEST_NAME=TestFWCoreSkeletons
+
+# Create a new CMSSW dev area
+OLD_CMSSW_BASE=$CMSSW_BASE
+rm -rf $SCRAM_TEST_NAME
+mkdir $SCRAM_TEST_NAME
+cd $SCRAM_TEST_NAME
+scram -a $SCRAM_ARCH project $CMSSW_VERSION
+cd $CMSSW_VERSION/src
+eval `scram run -sh`
+git cms-init
+
+# Copy FWCore/Skeletons in cause unit test is run during PR tests which contains changes for FWCore/Skeletons
+if [ -d $OLD_CMSSW_BASE/src/FWCore/Skeletons ] ; then
+    mkdir FWCore
+    cp -r $OLD_CMSSW_BASE/src/FWCore/Skeletons FWCore/Skeletons
+    scram build
+fi
+
+# Create TestSubsystem for testing the scripts
+mkdir TestSubsystem
+
+# Test output of scripts that can be easily compiled
+pushd TestSubsystem
+
+mkdqmedanalyzer TestDQMAnalyzerStream example_stream
+mkdqmedanalyzer TestDQMAnalyzerGlobal example_global
+mkdqmedanalyzer -debug TestDQMAnalyzerStrDbg example_stream
+mkdqmedanalyzer -debug TestDQMAnalyzerGlbDbg example_global
+
+mkedanlzr TestEDAnalyzer
+mkedanlzr TestEDAnalyzerHisto example_histo
+
+mkedfltr TestEDFilter
+
+mkedprod TestEDProducer
+mkedprod TestEDProducerParticle example_myparticle
+
+mkrecord TestRecord
+
+popd
+git add TestSubsystem
+git commit -a -m "Add skeleton modules that can be easily compiled"
+scram b code-checks-all
+scram b code-format-all
+git diff
+if [ $(git diff --name-only | grep TestSubsystem | wc -l) -gt 0 ] ; then
+    echo "code-checks or code-format caused differences, skeleton templates need to be updated!"
+    exit 1
+fi
+scram build
+
+# Test output of scripts for which compilation would be difficult, but
+# code-checks/code-format still works
+pushd TestSubsystem
+
+mkdatapkg TestDataPackage
+
+mkskel TestSkeleton
+
+mktsel TestSelector
+mktsel TestSelectorTrack example_track
+
+popd
+git add TestSubsystem
+git commit -a -m "Other skeletong code"
+scram b code-checks-all
+scram b code-format-all
+git diff
+if [ $(git diff --name-only | grep TestSubsystem | wc -l) -gt 0 ] ; then
+    echo "code-checks or code-format caused differences, skeleton templates need to be updated!"
+    exit 1
+fi
+
+# Test output of scripts for which compilation, code-checks, and
+# code-format would be difficult
+pushd TestSubsystem
+mkedlpr TestEDLooper
+mkedlpr TestEDLooperRecData TestRecord TestData1 TestData2
+
+mkesprod TestESProducer
+mkesprod TestESProducerRec TestRecord
+mkesprod TestESProducerRecData TestRecord TestData
+
```

Comment:
@makortel , just to match the previous `pushd`, I would suggest to add `popd` here too

Reply:
Thanks, added